### DISCUSSION TITLE
Expand yt-dlnis to a Full-fledged Download Manager

### DISCUSSION THREAD

**Discussion by Brajesh3** [deniscerri/ytdlnis#510] [Ideas]
Hello,

First, I would like to express my appreciation for the fantastic work on yt-dlnis. The integration of yt-dlp and aria2c has made downloading audio and video content seamless and efficient.

## Request:
I would like to propose expanding the functionality of yt-dlnis to serve as a full-fledged download manager, rather than focusing solely on audio and video downloads.

## Details:
- Currently, yt-dlnis focuses primarily on downloading audio and video files.
- Expanding the app to handle various file types (e.g., documents, archives, images) would make it a versatile tool for users who need a comprehensive download manager.

## Benefits:
- This enhancement would attract a wider user base who require a multi-purpose download manager.
- Users would benefit from a unified application for managing all types of downloads, reducing the need to use multiple applications.
- The robust download capabilities of aria2c can be leveraged for a variety of file types, improving overall user experience.

Thank you for considering this feature request. I believe it would greatly enhance the usability and appeal of yt-dlnis.

**Comment by teddysulaimanGL** [comment]
I don't think it can be done. This app is using a certain library. And it doesn't support more things that yt-dlp offers. And I think that makes the app harder to maintain than it already is

**Comment by deniscerri** [comment]
@Brajesh3 I think its better if people just forked my application and made different flavors of it that implement other tools.
Handling that many executables in a single app is challenging and would make it too bloated.
Also the flavour idea, they could have a different download card that make sense for that kind of downloader. Because a good part of the the options in the download card wont apply to other downloaders, etc etc.
